src/Config/ObjectModel/BaseRunTimeConfigLoader.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.Converters;
+using Azure.DataApiBuilder.Config.NamingPolicies;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.DataApiBuilder.Config.ObjectModel
+{
+    public abstract class BaseRunTimeConfigLoader
+    {
+        protected readonly string? _connectionString;
+
+        public BaseRunTimeConfigLoader(string? connectionString = null)
+        {
+            _connectionString = connectionString;
+        }
+
+        /// <summary>
+        /// Returns RunTimeConfig. Needs to be implemented by the derived class.
+        /// </summary>
+        /// <param name="config">The loaded <c>RuntimeConfig</c>, or null if none was loaded.</param>
+        /// <returns>True if the config was loaded, otherwise false.</returns>
+        public abstract bool TryLoadKnownConfig([NotNullWhen(true)] out RuntimeConfig? config);
+
+        /// <summary>
+        /// Parses a JSON string into a <c>RuntimeConfig</c> object
+        /// </summary>
+        /// <param name="json">JSON that represents the config file.</param>
+        /// <param name="config">The parsed config, or null if it parsed unsuccessfully.</param>
+        /// <returns>True if the config was parsed, otherwise false.</returns>
+        public static bool TryParseConfig(string json, [NotNullWhen(true)] out RuntimeConfig? config, ILogger? logger = null, string? connectionString = null)
```

Comment:
Doesn't this make the assumption that all implementations of a loader are going to do JSON deserialization?

Reply:
This seemed more like a utlity method as it didnt operate on any class variables to Load a runtime config object from a json.  The same code can be used by all implementators of this base class. For the scenario where you dynamically initialize the runtimeconfig you could have a case where you are doing this only for the first time then serializing to a json storing it in a cache then on next call loading it in and it would be useful to have the tryparseconfig.

Reply:
I get that we should push the method up to the base class, since it's something that we'll need to implement for all the different providers. But my understanding was that the provider you are looking at won't be providing JSON so you're going to override this anyway, so we'd be best to have this as just an abstract method and we have one implementation that is doing JSON parsing (the current implementation) and another that is doing your parsing.

Reply:
We would be providing json possibily on subsequent calls if we decide to store a json of the runtimeconfig and store it in local cache. I thought the tryparseconfig method was very generic and pretty much what is to be expected with json parsing and hence kept the implementation there.